internal/util/parse/parse.go

```diff
@@ -38,8 +39,13 @@ func GitParseArgs(ctx context.Context, args []string) (Target, error) {
 		return g, nil
 	}
 
+	parsedUrl, err := url.Parse(args[0])
+	if err != nil {
+		return g, err
+	}
```

Comment:
I think this will break if user specifies url in the form of `git@github.com:GoogleContainerTools/kpt.git`

I looked at the code a little bit and realized looking for `.git` is more about whether the url uses `git@github.com:org/repo.git` vs http based url `https://github.com/GoogleContainerTools/kpt.git`

Reply:
To feel more confident about the change, It will be good to have test that tests GitParseArgs function and test it for three forms of pkg URLs `git@....`, `http://`,`github.com:` etc.

Reply:
I think you're right that it'll break. Since we're checking for `.git($|/)` now I don't see much use for url.Parse anymore so I took this out. 

Added tests for `GitParseArgs` to `parse_test.go` to check the different url types